### PR TITLE
Highlighting and tapping on user mentions

### PR DESCRIPTION
**Pull request by VIPlearner**
### 🎯 Goal

The goal of this PR is to add functionality that allows users to click/tap user mentions.
Issue link: https://github.com/GetStream/stream-chat-android/issues/5263

### 🛠 Implementation details

I basically scan the string for any user mentions using the @ symbol as a reference and checking the mentionedUser list to confirm the user and register the part of the string as clickable

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>

Screenshot 2024-09-15 at 08 53 48

</td>
<td>

Screenshot 2024-09-15 at 08 51 41

</td>
</tr>
</tbody>
</table>

### 🧪 Testing

<details>

<summary>Tag a user when sending a message and send. 
The tagged username will be clickable
However nothing will happen when clicked 
you could however check the logs to confirm that the tagged username is being clicked
</summary>

</details>

### CODE REVIEW

**Comment by aleksandar-apostolov** [comment]
Hello @VIPlearner,

Thank you for your valuable contribution. We truly appreciate your efforts and will carefully consider your suggestion. Before moving forward, we'll need to consult with our product team to determine if we can include this in our SDK, as we aim to maintain parity with other platforms.

Best regards

**Comment by aleksandar-apostolov** [comment]
Hey @VIPlearner 

Before proceeding further, please follow these steps to ensure code quality / styling and stability:

### Fix styling issues and update the API File:
Run the following commands to fix any styling issues and update the API file:
```bash
./gradlew apiDump spotlessApply
```
### Resolve `klint` Check Failures:
The klint check is currently failing. You can identify and fix the issues locally by running:
```bash
./gradlew detekt
```
### Ensure unit tests stability:
Verify that no existing unit tests are broken by running the tests. Address any failures before proceeding.

**Comment by VIPlearner** [comment]
Hi @aleksandar-apostolov Please I've updated the branch
Thanks

**Comment by aleksandar-apostolov** [comment]
Hi @VIPlearner & @IsaacNewton111 
Did you signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform)? 
As external contributors you need to do so before we can merge the PR. 

Thanks

**Comment by VIPlearner** [comment]
Hi @aleksandar-apostolov yes please
 I have signed the CLA that was one of the first things i did while going through the checklist thanks

**Comment by VIPlearner** [comment]
However @aleksandar-apostolov Please I have a question
I appreciate your help with the PR. I noticed that there’s some alignment you were considering with the product team, and I’d love to better understand the project’s direction so I can make more meaningful contributions in the future. Could you share any guidance on some of the immediate goals or priorities for the project?
I’ve enjoyed working with this project and would like to contribute more if possible.
Thanks

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt [hunk 1]
```diff
@@ -161,6 +161,9 @@ class MessagesActivity : BaseConnectedActivity() {
                     onUserAvatarClick = { user ->
                         Log.i("MessagesActivity", "user avatar clicked: ${user.id}")
                     },
+                    onTapUserMention = { user ->
```
Comment (kanat) [inline stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt:164]: Hey @VIPlearner,

Thank you for the input!
Would you mind renaming `onTapUserMention` into `onUserMentionClick` for the sake of consistency throughout the PR?

Comment (VIPlearner) [reply stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/MessagesActivity.kt]: Hi @kanat done

**Comment by aleksandar-apostolov** [comment]
> However @aleksandar-apostolov Please I have a question I appreciate your help with the PR. I noticed that there’s some alignment you were considering with the product team, and I’d love to better understand the project’s direction so I can make more meaningful contributions in the future. Could you share any guidance on some of the immediate goals or priorities for the project? I’ve enjoyed working with this project and would like to contribute more if possible. Thanks

Hey @VIPlearner 

Thanks for your question and for your efforts on the PR! I appreciate your interest in aligning with our goals. While there are ongoing discussions with the product team about new features, our continuous focus remains on ensuring a solid and stable foundation for the project. Any contributions towards refining the core features and improving stability are always valuable. Looking forward to seeing more of your work!

**Review by JcMinarro — APPROVED** [review]
LGTM
Fix the conflicts and we can merge it

**Review by kanat — APPROVED** [review]

**Comment by VIPlearner** [comment]
Done @JcMinarro

**Pull request merged**